File: reservation-frontend/src/app/core/models/salle/salle.ts
import { TarificationSalle } from './tarification-salle';
import { Service } from '../service/service';
import { Espace } from '../espace/espace'; // Ajuste le chemin selon ton arborescence

export class Salle {
  salleId?: number;
  nom?: string;
  capaciteMax?: number;
  description?: string;
  adresse?: string;
  ville?: string;
  telephone?: string;
  email?: string;

  // Relations
  espace?: Espace;
    tarifications?: TarificationSalle[] = [];
  services?: Service[] = [];

  constructor(
    salleId?: number,
    nom: string = '',
    capaciteMax: number = 0,
    espace?: Espace,
    tarifications: TarificationSalle[] = [],
    services: Service[] = []
  ) {
    if (salleId !== undefined) this.salleId = salleId;
    this.nom = nom;
    this.capaciteMax = capaciteMax;
    if (espace) this.espace = espace;
    this.tarifications = tarifications;
    this.services = services;
  }
}

File: Reservation/src/main/java/net/travel/reservation/entites/Salle.java
package net.travel.reservation.entites;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "salles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Salle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long salleId;

    @Column(nullable = false)
    private String nom;

    @Column(nullable = false)
    private Integer capaciteMax;

    @Column(length = 500)
    private String description;

    private String adresse;

    private String ville;

    private String telephone;

    @Column(unique = true)
    private String email;

    // ✅ Relation vers TarificationSalle
    @OneToMany(mappedBy = "salle", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @JsonIgnore
    @Builder.Default
    private List<TarificationSalle> tarifications = new ArrayList<>();
    // ✅ Relation vers Service
    @OneToMany(mappedBy = "salle", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @JsonIgnore
    @Builder.Default
    private List<Service> services = new ArrayList<>();

    // ✅ AJOUT : Relation bidirectionnelle vers User (Les employés/gestionnaires de la salle)
    @OneToMany(mappedBy = "salle", fetch = FetchType.LAZY)
    @JsonIgnore // Évite la boucle de référence JSON infinie lors de la sérialisation
    @Builder.Default
    private List<User> users = new ArrayList<>();
}